plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -0,0 +1,595 @@
+        ImmutableList.Builder<Type> columnTypes = ImmutableList.builderWithExpectedSize(expectedSize);
+        for (ColumnMetadata columnMetadata : tableMetadata.getColumns()) {
+            columns.add(getColumnDefinitionSql(session, columnMetadata, columnMetadata.getName()));
+            columnNames.add(columnMetadata.getName());
+            columnTypes.add(columnMetadata.getType());
+        }
+        String sql = buildCreateSql(schemaTableName, columns.build(), tableMetadata.getProperties());
+
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            execute(session, connection, sql);
+
+            return new IgniteOutputTableHandle(
+                    schemaTableName.getSchemaName(),
+                    schemaTableName.getTableName(),
+                    columnNames.build(),
+                    columnTypes.build(),
+                    Optional.empty(),
+                    IgniteTableProperties.getPrimaryKey(tableMetadata.getProperties()).isEmpty() ? Optional.of(IGNITE_DUMMY_ID) : Optional.empty());
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, "Error creating Ignite table", e);
+        }
+    }
+
+    private String buildCreateSql(SchemaTableName schemaTableName, List<String> columns, Map<String, Object> tableProperties)
+    {
+        ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+        ImmutableList.Builder<String> columnDefinitions = ImmutableList.builder();
+        columnDefinitions.addAll(columns);
+
+        List<String> primaryKeys = IgniteTableProperties.getPrimaryKey(tableProperties);
+        checkArgument(primaryKeys.size() < columns.size(), "Ignite table must have at least one non PRIMARY KEY column.");
+        if (primaryKeys.isEmpty()) {
+            columnDefinitions.add(quoted(IGNITE_DUMMY_ID) + " VARCHAR NOT NULL");
+            primaryKeys = ImmutableList.of(IGNITE_DUMMY_ID);
+        }
+        columnDefinitions.add("PRIMARY KEY (" + join(", ", primaryKeys.stream().map(this::quoted).collect(joining(", "))) + ")");
+
+        for (Map.Entry<String, Object> propertyEntry : tableProperties.entrySet()) {
+            String propertyKey = propertyEntry.getKey();
+            if (!PRIMARY_KEY_PROPERTY.equalsIgnoreCase(propertyKey)) {
+                throw new UnsupportedOperationException("Not support table property " + propertyKey);
+            }
+        }
+        List<String> withProperties = tableOptions.build();
+
+        String remoteTableName = quoted(null, schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        String createTableSql = format("CREATE TABLE %s (%s) ", remoteTableName, join(", ", columnDefinitions.build()));
+        // If the table properties all are default value
+        if (withProperties.isEmpty()) {
+            return createTableSql;
+        }
+        return createTableSql + format(" WITH \" %s \"", join(", ", withProperties));
```

Comment:
This line is not reachable in my understanding.

Reply:
👍 It was introduced by removing the support for the table properties. Removed the `withProperties`